tools/apiview/parsers/CONTRIBUTING.md

```diff
@@ -0,0 +1,103 @@
+- `BottomToken (bt)` : Data that closes out the node. Depending on the language this would include the closing curly brace and/or empty lines. You can simulate an empty line by adding an empty token (content token with no value) and a lineBreak token.
+- `Children (c)` : The nodes immediate children. For a namespace this would be classes, for a class this would be the class constructors and methods.
+
+### StructuredToken
+- `Value (v)` : The token value which will be dispalyed. Can be null.  Spacing tokens don't need to have value.
+- `Id (i)` : This is essentially whatever existing parser was asigning to the token DefinitionId. You dont have to assign a tokenId.
+- `Kind (k)` : An enum
+  - `Content` Specifies that the token is content
+  - `LineBreak` Space token indicating switch to new line.
+  - `NoneBreakingSpace` Regular single space
+  - `TabSpace` 4 NoneBreakingSpaces
+  - `ParameterSeparator` Use this between method parameters. Depending on user setting this would result in a singlespace or new line
+  - `Url` A url token should have `LinkText` property and the url/link should be the token value.
+  All tokens should be content except for spacing tokens and url. ParameterSeparator should be used between method or function parameters. 
+- `Tags (t)` : Use this for opt in or opt out boolean properties.The currently supported tags are 
+  - `SkippDiff`  Indicate that a token should not be used in computatation of diff.
+  - `Deprecated` Mark a token as deprecated
+- `Properties (p)` : Properties of the token.
+  - `GroupId` : `doc` to group consecutive comment tokens as documentation.
+  - `NavigateToId` Id for navigating to where the node where the token is defined. Should match the definitionId of the node.
+- `RenderClasses (rc)` : Add css classes for how the tokens will be rendred. Classes currently being used are `text` `keyword` `punc` `tname` `mname` `literal` `sliteral` `comment` Feel free to add your own custom class. Whatever custom classes you use please provide us the appriopriate css for the class so we can update APIView.
+
+Json property names for each property is shown in brackets.
+
+Dont worry about indentation that will be handeled by the tree structure, unless you want to have indentation between the tokens then use `TabSpace` token kind.
+
+If your packages contains multiple assemblies then you will have multiple trees with multiple roots. Assign the final parsed value to a `List<APITreeNode> APIForest` property of the `CodeFile`.
+
+Serialize the generated code file to JSON. Try to make the json as small as possible by ignoring null values and empty collections, and using the abbreviated names of theproperties as the Json property name
```

Comment:
Again, this should go into a "Serialization" section. Also, don't you _have to_ use the abbreviations, or are they optional? It is more work on the parser side to generate these abbreviations.

Reply:
I have added Serialization section.